lib/core/form/components/widgets/container/container.widget.ts

```diff
@@ -57,31 +59,60 @@ export class ContainerWidgetComponent extends WidgetComponent implements OnInit,
     ngOnInit() {
         if (this.field) {
             this.content = new ContainerWidgetComponentModel(this.field);
+            this.getNumberOfColumnsFromTheBiggestBetweenJsonAndColumnsLengthOrOne();
+            this.fields = this.getFields();
         }
     }
 
+    private getNumberOfColumnsFromTheBiggestBetweenJsonAndColumnsLengthOrOne() {
+        this.numberOfColumns = (this.content.json?.numberOfColumns || 1) > (this.content.columns?.length || 1) ?
+            (this.content.json?.numberOfColumns || 1) :
+            (this.content.columns?.length || 1);
+    }
+
     /**
      * Serializes column fields
      */
-    get fields(): FormFieldModel[] {
+    private getFields(): FormFieldModel[] {
         const fields = [];
+        const toBeComputed = [];
+        const rowspanOffset = [];
+        let size = 0;
+        for (let i = 0; i < this.numberOfColumns; i++) {
+            toBeComputed.push(this.content.columns[i]?.fields?.length || 0);
+            rowspanOffset[i] = 0;
+            size += (this.content.columns[i]?.fields?.length || 0);
+        }
 
-        let rowContainsElement = true,
-            rowIndex = 0;
-
-        while (rowContainsElement) {
-            rowContainsElement = false;
-            for (let i = 0; i < this.content.columns.length; i++ ) {
-                const field = this.content.columns[i].fields[rowIndex];
-                if (field) {
-                    rowContainsElement = true;
+        for (let i = 0; i < size; i++) {
+            let fieldExist = false;
+            let columnIndex = 0;
+            while (columnIndex < this.numberOfColumns) {
+                let field;
```

Comment:
can you group some of the parts in the while with some methods?

Reply:
Done! Thanks.